src/riak_ql_ddl.erl

```diff
@@ -179,22 +180,44 @@ get_local_key(?DDL{table = T}=DDL, Obj)
     Mod = make_module_name(T),
     get_local_key(DDL, Obj, Mod).
 
+-spec desc_adjusted_key([any()], module(), ?DDL{}) ->
```

Comment:
I think all client code should call `negate_if_desc(lk_to_pk(Keys, Mod, DDL), Mod, DDL)` *or* `negate_if_desc(Keys, Mod, DDL)`.